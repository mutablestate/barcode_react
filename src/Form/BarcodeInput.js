import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// BarcodeInput
const BarcodeInput = ({ name, value, maxLength, onChange, status }) => {
  return (
    <Fragment>
      <div className="form-control">
        <label>Barcode Digits</label>
        <input
          type="text"
          name={name}
          value={value}
          onChange={onChange}
          maxLength={maxLength}
          data-test="BarcodeInput:input"
        />
        <span className="barcode-status" data-test="BarcodeInput:status">
          {status}
        </span>
      </div>
    </Fragment>
  );
};

BarcodeInput.propTypes = {
  /** The input tag name attribute value */
  name: PropTypes.string.isRequired,
  /** The input tag maxLength attribute value */
  maxLength: PropTypes.number.isRequired,
  /** The onChange event handler */
  onChange: PropTypes.func.isRequired,
  /** The input tag value attribute */
  value: PropTypes.string,
  /** Barcode status message to inform users of the current status */
  status: PropTypes.string.isRequired
};

export default BarcodeInput;
