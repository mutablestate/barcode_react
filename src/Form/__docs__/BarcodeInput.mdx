---
name: BarcodeInput
route: /forms/barcode-input
menu: Form
---

import { Playground, Props } from 'docz';
import BarcodeInput from '../BarcodeInput';

# BarcodeInput

A basic text input field with maxLength and status text.

## Props

<Props of={BarcodeInput} />

## Basic usage

<Playground>
  <BarcodeInput
    status={'Enter up to 19 digits'}
    name={'fieldName'}
    maxLength={19}
    onChange={() => {}}
  />
</Playground>
