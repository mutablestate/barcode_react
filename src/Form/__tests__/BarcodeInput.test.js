import React from 'react';
import renderer from 'react-test-renderer';

import BarcodeInput from '../BarcodeInput';

const requiredProps = {
  name: 'myFieldName',
  status: 'Enter up to 19 digits',
  maxLength: 19,
  onChange: jest.fn()
};

const optionalProps = {
  value: 'currentValue'
};

describe('BarcodeInput', () => {
  it('must render correctly with required props', () => {
    const component = renderer
      .create(<BarcodeInput {...requiredProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must render correctly with optional props', () => {
    const updatedProps = {
      ...requiredProps,
      ...optionalProps
    };
    const component = renderer
      .create(<BarcodeInput {...updatedProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
