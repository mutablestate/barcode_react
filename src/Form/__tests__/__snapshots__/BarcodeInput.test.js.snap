// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarcodeInput must render correctly with optional props 1`] = `
<div
  className="form-control"
>
  <label>
    Barcode Digits
  </label>
  <input
    data-test="BarcodeInput:input"
    maxLength={19}
    name="myFieldName"
    onChange={[MockFunction]}
    type="text"
    value="currentValue"
  />
  <span
    className="barcode-status"
    data-test="BarcodeInput:status"
  >
    Enter up to 19 digits
  </span>
</div>
`;

exports[`BarcodeInput must render correctly with required props 1`] = `
<div
  className="form-control"
>
  <label>
    Barcode Digits
  </label>
  <input
    data-test="BarcodeInput:input"
    maxLength={19}
    name="myFieldName"
    onChange={[MockFunction]}
    type="text"
  />
  <span
    className="barcode-status"
    data-test="BarcodeInput:status"
  >
    Enter up to 19 digits
  </span>
</div>
`;
